src/meetingsession/DefaultMeetingSession.ts

```diff
@@ -49,9 +46,14 @@ export default class DefaultMeetingSession implements MeetingSession, Destroyabl
 
     CSPMonitor.addLogger(this._logger);
     CSPMonitor.register();
-
-    this.setupEventReporter(configuration, logger, eventReporter);
-    this._deviceController = deviceController;
+    if (eventController) {
+      this._eventController = eventController;
+    } else {
+      this._eventController = new DefaultEventController(configuration, logger);
+    }
+    if (!deviceController.getEventController()) {
+      deviceController.setEventController(this.eventController);
```

Comment:
This seems odd, and I'm not sure its totally necessary given how few builders are going to be injecting these (would be nice to get rid of the getter/setter on the device controller). But maybe this will look different when we decouple device controller so I won't think about it too much. But i wouldn't mind a `TODO: Consider whether this is necessary`.

Reply:
This is a bit of a odd case, but because we want to decouple the `DeviceController` from the `MeetingSession` we need it to be able to have it's own `EventController`, but we also want for the `DeviceController` to be able to easily use the same `EventController` as the `MeetingSession` meaning a builder shouldn't have to create their own `EventController` for this to be possible so there's a need to be able to set the controller after creation. However, there is also a possibility that a builder might want to be able to have different `EventController`'s for `DeviceController` and `MeetingSession` so we need to also check to make sure there isn't already one on `DeviceController`

Reply:
It seems to me like those may be phantom use cases, or a case where we are trying a bit too hard to save builders who are already doing complicated things (like injecting) a single line of code or requirement, but I will still wait until the device controller stuff before nagging more, since I'm not really familiar with how that will look.

Reply:
Yeah we can discuss more about this once we see what the `DeviceController` looks like decoupled.